ui/src/app/views/workspace/component.ts

```diff
@@ -283,6 +292,38 @@ export class WorkspaceComponent implements OnInit, OnDestroy {
     this.router.navigate(['share'], {relativeTo : this.route});
   }
 
+  selectANotebook(notebook): void {
+    let loopedNotebook = 0;
+    this.checkColumnNotebook = true;
+    this.enablePushNotebookBtn = false;
+
+    for (const file of this.notebookList) {
+      loopedNotebook++;
+      if (!file.selected) {
+        this.checkColumnNotebook = false;
+        break;
+      }
+    }
+    if (loopedNotebook === 1) {
```

Comment:
Sorry, maybe I don't understand what you're trying to do here - I'm even more confused by this function now. Is there something incorrect in the logic of the 2 lines of (IMO) much more readable code I gave in my comment? Rewrote to capture updated code:

```
this.checkColumnNotebook = this.notebookList.all(n => n.selected);
this.enablePushNotebookBtn = this.notebookList.some(n => n.selected);
```

Reply:
fixed